llvm/lib/TargetParser/Host.cpp

```diff
@@ -1845,6 +1845,7 @@ bool sys::getHostCPUFeatures(StringMap<bool> &Features) {
   Features["prefetchi"]  = HasLeaf7Subleaf1 && ((EDX >> 14) & 1);
   Features["usermsr"]  = HasLeaf7Subleaf1 && ((EDX >> 15) & 1);
   Features["avx10.1-256"] = HasLeaf7Subleaf1 && ((EDX >> 19) & 1);
+  Features["apxf"] = HasLeaf7Subleaf1 && ((EDX >> 21) & 1);
```

Comment:
I don't think we have a `apxf` feature in LLVM, should it be `egpr`?

Reply:
This patch defined one in clang through llvm/include/llvm/TargetParser/X86TargetParser.def. I think it's ok to do so.

Reply:
My understanding is "apxf" here is used by front-end for FMV, right?
What would happen if there is "apxf" in the IR after this patch?

Reply:
> My understanding is "apxf" here is used by front-end for FMV, right?

Yes, not only FMV, but also __builtin_cpu_supports.
> What would happen if there is "apxf" in the IR after this patch?

if llc read "apxf" in attribute, I think a warning will be given like:
```
'+apxf' is not a recognized feature for this target (ignoring feature)
```
But such attribute won't be generated from frontend, because we didn't add logics for "apxf" in X86TargetInfo::hasFeature.

Reply:
> if llc read "apxf" in attribute, I think a warning will be given like:
> 
> ```
> '+apxf' is not a recognized feature for this target (ignoring feature)
> ```

Did you check the behavior w/ this patch?

Reply:
Emm, I happened to find an issue with this patch.
```
$ clang -march=native foo.c -S -emit-llvm 
'-apxf' is not a recognized feature for this target (ignoring feature)
```
It turns out even not adding logics in X86TargetInfo::hasFeature it will still return false for "apxf". I also found "-egpr" is not generated. How did you handle this?

Reply:
I think you cannot actually enable APX features when using `__attribute__((__target__("apxf")))` either.

Reply:
Yeah, that's another problem. For attribute/target, it seems more reasonable to enable as "egpr" instead of "apxf" because apx needs such fine-grained enabling. We need to discuss with gcc on this.